src/scripts.js

```diff
@@ -1,153 +1,157 @@
+    domUpdates.connectWithClassList('remove', 'hidden', event, favoriteButton);
+    domUpdates.displayCards(cookbook.recipes, cardArea);
+    getFavorites();
+  }
 }
 
-function displayIngredients(recipeObject) {
-  const ingredientsSpan = document.querySelector('.ingredients');
-  const instructionsSpan = document.querySelector('.instructions');
-  recipeObject.ingredients.forEach(ingredient => {
-    ingredientsSpan.insertAdjacentHTML('afterbegin', `<ul><li>
-    ${ingredient.quantity.amount.toFixed(2)} ${ingredient.quantity.unit}
-    ${ingredient.name}</li></ul>
-    `)
-    })
-    recipeObject.instructions.forEach(instruction => {
-        instructionsSpan.insertAdjacentHTML('beforebegin', `<li>
-    ${instruction.instruction}</li>
-    `)
-    })
+function displayDirections(event) {
+  domUpdates.connectWithClassList('remove', 'hidden', event, favoriteButton)
+  let newRecipeInfo = cookbook.recipes.find(recipe => recipe.id === Number(event.target.id))
+  let recipeObject = new Recipe(newRecipeInfo, ingredientData);
+  let cost = recipeObject.calculateCost();
+  let dollarCost = (cost / 100).toFixed(2);
+  domUpdates.connectWithClassList('add', 'all', event, cardArea);
+  domUpdates.populateRecipeCard(cardArea, recipeObject, dollarCost, ingredientData);
 }
 
 function getFavorites() {
   if (user.favoriteRecipes.length) {
     user.favoriteRecipes.forEach(recipe => {
-      document.querySelector(`.favorite${recipe.id}`).classList.add('favorite-active')
+      let recipeID = document.querySelector(`.favorite${recipe.id}`);
+      domUpdates.connectWithClassList('add', 'favorite-active', event, recipeID);
     })
   }
 }
 
-function populateCards(recipes) {
-  cardArea.innerHTML = '';
-  if (cardArea.classList.contains('all')) {
-    cardArea.classList.remove('all')
-  }
-  insertCards(recipes);
-  getFavorites();
-}
-
-function insertCards(cardsToDisplay) {
-  cardsToDisplay.forEach(recipe => {
-    cardArea.insertAdjacentHTML('afterbegin', `<div id='${recipe.id}'
-  class='card'>
-      <header id='${recipe.id}' class='card-header'>
-        <label for='add-button' class='hidden'>Click to add recipe</label>
-        <button id='${recipe.id}' aria-label='add-button' class='add-button card-button'>
-          <img id='${recipe.id} favorite' class='add'
-          src='https://image.flaticon.com/icons/svg/32/32339.svg' alt='Add to
-          recipes to cook'>
-        </button>
-        <label for='favorite-button' class='hidden'>Click to favorite recipe
-        </label>
-        <button id='${recipe.id}' aria-label='favorite-button' class='favorite favorite${recipe.id} card-button'></button>
-      </header>
-        <span id='${recipe.id}' class='recipe-name'>${recipe.name}</span>
-        <img id='${recipe.id}' tabindex='0' class='card-picture'
-        src='${recipe.image}' alt='click to view recipe for ${recipe.name}'>
-  </div>`)
+function displaySearchRecipes(event) {
```

Comment:
I noticed that this functionality wasn't working when I tried it on my local device.